.github/workflows/common.sh

```diff
@@ -32,36 +36,44 @@ function get_ebuild_filename() {
 }
 
 function prepare_git_repo() {
-  local our_remote_url
-
+  # the original coreos-overlay repo outside the SDK container
   git config user.name "${BUILDBOT_USERNAME}"
   git config user.email "${BUILDBOT_USEREMAIL}"
   git reset --hard HEAD
   git fetch origin
+
   git checkout -B "${BASE_BRANCH}" "origin/${BASE_BRANCH}"
-  our_remote_url=$(git remote get-url origin)
 
-  # setup overlay repo inside SDK too (be fork friendly)
-  git -C "${SDK_OUTER_SRCDIR}/third_party/coreos-overlay" remote add our_remote "${our_remote_url}"
-  git -C "${SDK_OUTER_SRCDIR}/third_party/coreos-overlay" fetch our_remote
-  git -C "${SDK_OUTER_SRCDIR}/third_party/coreos-overlay" checkout -B "${BASE_BRANCH}" "our_remote/${BASE_BRANCH}"
+  # inside the SDK container
+  git -C "${SDK_OUTER_SRCDIR}/third_party/coreos-overlay" config \
+    user.name "${BUILDBOT_USERNAME}"
+  git -C "${SDK_OUTER_SRCDIR}/third_party/coreos-overlay" config \
+    user.email "${BUILDBOT_USEREMAIL}"
 }
 
 # caller needs to set pass a parameter as a branch name to be created.
 function checkout_branches() {
   TARGET_BRANCH=$1
+  CHECKOUT_SCRIPTS=$2
 
   [[ -z "${TARGET_BRANCH}" ]] && echo "No target branch specified. exit." && return 1
 
-  git -C "${SDK_OUTER_SRCDIR}/scripts" checkout -B "${BASE_BRANCH}" "github/${BASE_BRANCH}"
-  git -C "${SDK_OUTER_SRCDIR}/third_party/portage-stable" checkout -B "${BASE_BRANCH}" "github/${BASE_BRANCH}"
-
-  if git -C "${SDK_OUTER_SRCDIR}/third_party/coreos-overlay" show-ref "remotes/our_remote/${TARGET_BRANCH}"; then
-    echo "Target branch already exists. exit.";
-    return 1
+  # Check out the scripts repo only if CHECKOUT_SCRIPTS == true, due to
+  # a corner case of its LTS-2021 branch does not have run_sdk_container.
```

Comment:
When we merge https://github.com/flatcar-linux/scripts/pull/388 this won't be needed